### PR TITLE
[docs] #85: Add UML diagrams, create a new chapter about objects

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

- a new chapter that covers the info about objects in a blockchain
- UML diagrams with the hierarchy (I've only added a couple, not covering everything in the hierarchy yet. This is just to make sure this is the right format)
- the topic on metadata from the advanced section was moved to this new chapter
- a separate topic with the info about assets was created within this same chapter (that info used to be copy-pasted between lang-specific guides)
- minor changes in the topic on permissions

### CODE REVIEW

File: src/guide/diagrams-src/domain-account-asset.puml
Comment (appetrosyan) [inline src/guide/diagrams-src/domain-account-asset.puml:34]: Consider adding newlines at end of file. This isn't strictly necessary, but every other file has a newline at the end.

**Review by appetrosyan — APPROVED** [review]

File: src/guide/advanced/permissions.md
Comment (WRRicht3r) [inline src/guide/advanced/permissions.md:344]:
```suggestion
roles can be granted using. the `Grant` instruction and revoked using the `Revoke`
```

File: src/guide/advanced/permissions.md
Comment (WRRicht3r) [inline src/guide/advanced/permissions.md:347]:
```suggestion
Before granting a role to an account, the role should be registered first.
```

File: src/guide/advanced/queries.md
Comment (WRRicht3r) [inline src/guide/advanced/queries.md:120]:
```suggestion
[metadata](../objects/metadata.md), as the metadata key is used to sort
```

File: src/guide/objects/hierarchy.md
Comment (WRRicht3r) [inline src/guide/objects/hierarchy.md:35]:
```suggestion
of Iroha. In the stable version, permission tokens aren't stored within
```

File: src/guide/objects/metadata.md
Comment (WRRicht3r) [inline src/guide/objects/metadata.md:61]:
```suggestion
[role](../advanced/permissions.md#permission-groups-roles) for accessing the
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

**Review by 6r1d — APPROVED** [review]
LGTM, ping me after addressing Willam's changes if GitHub requires a re-approval.

File: src/guide/advanced/permissions.md
Comment (outoftardis) [reply src/guide/advanced/permissions.md:344]:
```suggestion
roles can be granted using the `Grant` instruction and revoked using the `Revoke`
```

**Review by WRRicht3r — APPROVED** [review]

**Pull request merged**
